packages/tasit-action/src/ethers.test.js

```diff
@@ -83,6 +84,33 @@ describe("ethers.js", () => {
     ]);
   });
 
+  // Note: BUG on ethers.removeAllListeners functions
+  // See more: https://github.com/ethers-io/ethers.js/issues/391
+  it.skip("ethers.js removeAllListeners bug - listeners remain registered", async () => {
+    const eventFakeFn = sinon.fake();
+
+    const sentTx = await contract.setValue("hello world");
+
+    let eventListener;
+
+    await new Promise(function(resolve, reject) {
+      eventListener = () => {
+        eventFakeFn();
+        resolve();
+      };
+      contract.on("ValueChanged", eventListener);
+    });
+
+    // Not working
+    contract.removeAllListeners("ValueChanged");
+    // Working
+    // contract.removeListener("ValueChanged", eventListener);
+
+    expect(eventFakeFn.called).to.be.true;
```

Comment:
This is called because the listener is triggered even though the transaction was already sent? Should we send the transaction after registering the listener for it to be clearer?

Reply:
A) Yes
I think that it's more clear to me how `ether.js` events listening works internally  (related to https://github.com/tasitlabs/TasitSDK/issues/110). 

B) That's works because new block will be mined after the transaction and then that block will check on next `ethers`'s pool call (because that Promise wait is necessary). If do you want, I can rewrite this to change the order. It won't be worked without rewriting because Promise will be waiting forever.

Reply:
B) I've made this attempt:
- Waiting for 2 * pollingInterval between blocks mining. Simulating how that's works on real blockchain (pollingInterval < blockTime).
That way I've hopefully with that will be easier to poll routine catch de event and then write something like that:

```javascript
    const sentTx = await contract.setValue(newValue);
    contract.on("ValueChanged", listener);
    // after mining, waiting for enough time to doPoll runs. 
    await mineBlocks(provider, 1); 
```
This worked sometimes only if blocks to mining value was increased.
I think that the poll routine isn't determinist (a lot of promise.then()).
Because of that, using `waitForEvent` functions seems the most reliable way to write contract events tests (Note: As same as ethers.js does in their tests).

Reply:
What do you mean by `blocks to mining value`?

Reply:
Hmm yes, heavily using `waitForEvent` because of that seems reasonable

Reply:
Would you suggest any changes to the ethers.js poll routine based on this?

Reply:
> What do you mean by blocks to mining value?

How many blocks to mine.

> Would you suggest any changes to the ethers.js poll routine based on this?

I've think that maybe a function like `waitForPollFinish` will be useful. Should I create an issue sharing that situation?

Reply:
> How many blocks to mine

Makes sense now!

> ... issue ...

Yep, I think it’s a good idea to be good contributors and regular providers of feedback to ethers.js

Reply:
Done https://github.com/ethers-io/ethers.js/issues/393